src/PFire.Core/Session/XFileClientManager.cs

```diff
@@ -0,0 +1,65 @@
+﻿using PFire.Core.Protocol.Interfaces;
+using PFire.Database;
+using PFire.Session;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PFire.Core.Session
+{
+    internal sealed class XFileClientManager : IXFireClientManager
+    {
+        // the Dictionary<> is thread safe when read, but not when modified
+        private readonly Dictionary<Guid, XFireClient> _sessions;
```

Comment:
I don't think you'll need to worry about locking if you use https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-5.0 instead

Reply:
Good call